### PR TITLE
draft-reviews

### PR DESCRIPTION
**Pull request by arsenyosemin**
Adds an autosave as well as a save feature to reviews.
Reviews start in a "draft" state where they can be saved without having all scores submitted.
On "submit" the review and all criteria reviews exit draft and validations are run.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@
     t.string "name", null: false
     t.text "body"
     t.string "record_type", null: false
-    t.bigint "record_id", null: false
+    t.integer "record_id", null: false
```
[the remark this answers is not in the document]
Comment (arsenyosemin) [reply db/schema.rb:22]: oh strange. yeah wonder what is up with that. will revert

File: app/controllers/grant_submissions/submissions/reviews_controller.rb [hunk 2]
```diff
@@ -66,13 +66,16 @@ def update
...
+            format.html do
+              set_redirect_path
+              redirect_to @redirect_path,
+                          notice: 'Review was successfully updated.'
+            end
             format.json { render :show, status: :ok, location: @review }
           else
+            format.js   { render json: { errors: @review.errors.full_messages, success: false } }
             flash.now[:alert] = @review.errors.full_messages
-            build_criteria_reviews
```
Comment (jseraf) [inline app/controllers/grant_submissions/submissions/reviews_controller.rb:75]: should this have been removed?

File: app/helpers/grant_submissions/submissions_helper.rb [hunk 3]
```diff
@@ -2,15 +2,15 @@ module GrantSubmissions::SubmissionsHelper
   def review_data(submission)
     if submission.reviews.any?
       reviews = submission.reviews.to_a
-      OpenStruct.new(completed_review_count: reviews.count{ |review| !review.overall_impact_score.nil? },
-                     assigned_review_count:  reviews.count,
+      OpenStruct.new(completed_review_count: reviews.count(&:is_complete?),
```
Comment (jseraf) [inline app/helpers/grant_submissions/submissions_helper.rb:5]: could use the Review `completed` scope  here

**Review by jseraf — COMMENTED** [review]
Could you also add a draft review to the [scoring related submission model specs ](https://github.com/NUARIG/competitions/blob/develop/spec/models/grant_submission_submission_spec.rb#L159)

File: app/controllers/grant_submissions/submissions/reviews_controller.rb (hunk 2 above)
Comment (arsenyosemin) [reply app/controllers/grant_submissions/submissions/reviews_controller.rb:75]: I added it back, but build_criteria_reviews already runs when entering the edit page in the first place, so it seemed duplicated.

**Pull request merged**